### ISSUE TITLE
[REVIEW]: thresholdmodeling: A Python package for modeling excesses over a threshold using the Peak-Over-Threshold Method and the Generalized Pareto Distribution

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2013]
**Submitting author:** @iagolemos1 (<a href="http://orcid.org/0000-0002-5829-7711">Iago Lemos</a>)
**Repository:** <a href="https://github.com/iagolemos1/thresholdmodeling" target ="_blank">https://github.com/iagolemos1/thresholdmodeling</a>
**Version:** v0.0.1
**Editor:** @drvinceknight
**Reviewer:** @bahung, @kellieotto
**Archive:** <a href="https://doi.org/10.5281/zenodo.3661338" target="_blank">10.5281/zenodo.3661338</a>

## Status

[![status](https://joss.theoj.org/papers/ed29ca0eb021662682a366adffb87f6c/status.svg)](https://joss.theoj.org/papers/ed29ca0eb021662682a366adffb87f6c)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/ed29ca0eb021662682a366adffb87f6c"><img src="https://joss.theoj.org/papers/ed29ca0eb021662682a366adffb87f6c/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/ed29ca0eb021662682a366adffb87f6c/status.svg)](https://joss.theoj.org/papers/ed29ca0eb021662682a366adffb87f6c)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/iagolemos1/thresholdmodeling" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@bahung & @kellieotto, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @drvinceknight know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/iagolemos1/thresholdmodeling">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@iagolemos1) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/iagolemos1/thresholdmodeling">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@iagolemos1) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @bahung, @kellieotto it looks like you're currently assigned to review this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1007/978-1-4471-3675-0 is OK
- 10.1016/S0309-1708(02)00056-8 is OK
- 10.1017/CBO9780511529443 is OK
- 10.6028/jres.099.028 is OK
- 10.5194/os-2016-47 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02013/joss.02013/10.21105.joss.02013.pdf)

**Comment by iagolemos1** [comment]
I made some minimal corrections on the paper.

**Comment by iagolemos1** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02013/joss.02013/10.21105.joss.02013.pdf)

**Comment by iagolemos1** [comment]
@whedon generate pdf

**Comment by iagolemos1** [comment]
Below, I tried to consider all the changes suggested by @bahung.

**Comment by iagolemos1** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02013/joss.02013/10.21105.joss.02013.pdf)

**Comment by iagolemos1** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02013/joss.02013/10.21105.joss.02013.pdf)

**Comment by iagolemos1** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02013/joss.02013/10.21105.joss.02013.pdf)

**Comment by iagolemos1** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02013/joss.02013/10.21105.joss.02013.pdf)

**Comment by kellieotto** [comment]
@drvinceknight I recommend the package for approval. @iagolemos1 has responded to all of my comments.

**Comment by iagolemos1** [comment]
@kellieotto Thanks!

**Comment by iagolemos1** [comment]
@kellieotto Thanks for the support and to help my paper to get better!

**Comment by drvinceknight** [comment]
Thank you @kellieotto, I believe we are no waiting on @bahung to finish their review?

**Comment by iagolemos1** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02013/joss.02013/10.21105.joss.02013.pdf)

**Comment by drvinceknight** [comment]
@bahung apologies for the nag but any idea when you might be able to complete your review?

**Comment by bahung** [comment]
Hi @drvinceknight,

I thought you have approved as you suggested with @kellieotto in [the previous comment](https://github.com/openjournals/joss-reviews/issues/2013#issuecomment-578671696) 

TBH, I do not fully understand how do you organise this review by saying that way, so I stop taking further review steps, despite it seems to fail to initiate the package without R_USER and R_HOME being defined in the system paths.

**Comment by drvinceknight** [comment]
> I thought you have approved as you suggested with @kellieotto in [the previous comment](https://github.com/openjournals/joss-reviews/issues/2013#issuecomment-578671696)


Sorry I think there was a typo in that comment, I meant to write: `we are now waiting on @bahung to finish their review?`.

> TBH, I do not fully understand how do you organise this review by saying that way, so I stop taking further review steps, despite it seems to fail to initiate the package without R_USER and R_HOME being defined in the system paths.

Apologies I do not understand. If you're having problems installing the software this would be useful to know in your review.

We do need two reviews, if you're able to complete the review that would be great but if not please let me know and I can search for another reviewer.

**Comment by iagolemos1** [comment]
Hi @bahung, I think it happens when rpy2 package was not successfully or maybe you are having problems with which environment you are using. Just let me know what is happening then we can solve the problem :D 
The package should work successfully like it was for @kellieotto.

**Comment by iagolemos1** [comment]
I recommend uninstalling everything you did, then follow the steps In the [README](https://github.com/iagolemos1/thresholdmodeling).
```
conda activate my_env
conda install r
conda install -c r rpy2=2.9.4
pip install thresholdmodeling
```
Now I am realizing, this problem you had must be related to the R base for the software.

**Comment by bahung** [comment]
> > I thought you have approved as you suggested with @kellieotto in [the previous comment](https://github.com/openjournals/joss-reviews/issues/2013#issuecomment-578671696)
> 
> Sorry I think there was a typo in that comment, I meant to write: `we are now waiting on @bahung to finish their review?`.
> 
> > TBH, I do not fully understand how do you organise this review by saying that way, so I stop taking further review steps, despite it seems to fail to initiate the package without R_USER and R_HOME being defined in the system paths.
> 
> Apologies I do not understand. If you're having problems installing the software this would be useful to know in your review.
> 
> We do need two reviews, if you're able to complete the review that would be great but if not please let me know and I can search for another reviewer.

No problem, sorry for the misunderstanding. I will proceed as usual.

**Comment by bahung** [comment]
@drvinceknight I recommend the paper to be published as the author @iagolemos1 had responded to and addressed my issues.

**Comment by iagolemos1** [comment]
Thank you @bahung, for helping me all the support.

**Comment by drvinceknight** [comment]
@whedon generate proof

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by drvinceknight** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1007/978-1-4471-3675-0 is OK
- 10.1016/S0309-1708(02)00056-8 is OK
- 10.1017/CBO9780511529443 is OK
- 10.6028/jres.099.028 is OK
- 10.5194/os-2016-47 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by drvinceknight** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02013/joss.02013/10.21105.joss.02013.pdf)

**Comment by drvinceknight** [comment]
@iagolemos1 some minor typos: https://github.com/iagolemos1/thresholdmodeling/pull/7

**Comment by iagolemos1** [comment]
Merged the requests! Thanks @drvinceknight

**Comment by iagolemos1** [comment]
@drvinceknight Now I have to archive the repository in Zenodo, right?

**Comment by drvinceknight** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02013/joss.02013/10.21105.joss.02013.pdf)

**Comment by drvinceknight** [comment]
> @drvinceknight Now I have to archive the repository in Zenodo, right?

That's right: and pleas report the DOI in the review thread. Could you make sure the Zenodo deposit has the correct metadata (title and author list).


Can you also confirm the version of the software please?

**Comment by iagolemos1** [comment]
Version 0.0.1.

**Comment by iagolemos1** [comment]
DOI in Zenodo:
10.5281/zenodo.3661338

**Comment by drvinceknight** [comment]
@whedon set 10.5281/zenodo.3661338 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.3661338" target="_blank">10.5281/zenodo.3661338</a> is the archive.

**Comment by drvinceknight** [comment]
@openjournals/joss-eics this paper is now ready to be accepted :) 👍 🎆 


Thank you @bahung and @kellieotto  for your time and effort reviewing the work. :+1:

**Comment by danielskatz** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1007/978-1-4471-3675-0 is OK
- 10.1016/S0309-1708(02)00056-8 is OK
- 10.1017/CBO9780511529443 is OK
- 10.6028/jres.099.028 is OK
- 10.5194/os-2016-47 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1294

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1294, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by danielskatz** [comment]
👋 @iagolemos1  - please merge https://github.com/iagolemos1/thresholdmodeling/pull/8 or let me know what parts of it you disagree with.  Also, there is at least one case error in the .bib file, a pareto that should be {P}areto.  Please check the bib entries as built in the pdf carefully, and make any other needed changes as well.

**Comment by iagolemos1** [comment]
Hi @danielskatz. I merged the changes and made the changes in the .bib file. Thanks!

**Comment by danielskatz** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02013/joss.02013/10.21105.joss.02013.pdf)

**Comment by danielskatz** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1007/978-1-4471-3675-0 is OK
- 10.1016/S0309-1708(02)00056-8 is OK
- 10.1017/CBO9780511529443 is OK
- 10.6028/jres.099.028 is OK
- 10.5194/os-2016-47 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1295

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1295, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by danielskatz** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1226940677067616257) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/1296
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.02013](https://doi.org/10.21105/joss.02013)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? notify your editorial technical team...

**Comment by danielskatz** [comment]
Thanks to @bahung, @kellieotto for reviewing!
And @drvinceknight for editing!
And congratulations to @iagolemos1 and co-authors!

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02013/status.svg)](https://doi.org/10.21105/joss.02013)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02013">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02013/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02013/status.svg
   :target: https://doi.org/10.21105/joss.02013
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02013/status.svg)](https://doi.org/10.21105/joss.02013)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss

**Comment by iagolemos1** [comment]
@danielskatz I tryied to reach the paper using the DOI and it's appearing "File not foud". It's because the DOI isn't ready yet? 
![image](https://user-images.githubusercontent.com/57317828/74181061-527f6800-4c1f-11ea-99f8-0f5a7b46f310.png)

**Comment by danielskatz** [comment]
The link https://doi.org/10.21105/joss.02013 works fine for me (and did before I closed the issue) - it might just be an issue of how new DOIs resolve and how this information is distributed.  Please try again in a little bit and if it continue to not work for you, let us know.

As I look at this more carefully, the DOI is resolving for you, and bringing you to the right page.  The problem you are having is that the PDF at that page is not displaying, which is likely a web caching issue.  Again, however, waiting a little bit should work.

**Comment by iagolemos1** [comment]
Ok, thanks!
